### ISSUE TITLE
AQL does not obey name/value predicates

### ISSUE THREAD

**Issue by freshehr** [ethercis/ethercis#95]
```
b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose amount description']/value/value as dose_amount,
    b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose timing description']/value/value as dose_timing,
```
The resultset returns the value for dose_amount in both fields. This is an important AQL construct which we need to support.

Example: compositionUid":"d26af59d-547a-4e18-817f-e7f83996d3da::vm01.ethercis.org::1"

**Comment by chevalleyc** [comment]
Hi @freshehr yes indeed. Originally, I thought about these AQL enhancement in a short term timeframe (see [roadmap](https://github.com/ethercis/ethercis/blob/master/ethercis-roadmap.md)). This issue is somewhat related to partial node predicate/partial path etc. I am considering introducing a meta-data mechanism to support these type of queries efficiently. However, these short term objectives have been significantly postponed due to other current priorities and scarcity of resources.

**Comment by chevalleyc** [comment]
This type of queries now work f.e.
```
select
    a/uid/value, b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose amount description']/value/value as dose_amount,
   b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose timing description']/value/value as dose_timing   from EHR e[ehr_id/value='cd8abecd-9925-4313-86af-93aab4930eae']
    contains COMPOSITION a[openEHR-EHR-COMPOSITION.medication_list.v0]
        contains INSTRUCTION b_a[openEHR-EHR-INSTRUCTION.medication_order.v1]
   where b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose amount description']/value/value='Dose Amount'
```
also with multiple node predicates:

```
select
    a/uid/value, b_a/activities[at0001, 'Order']/description[at0002]/items[at0173, 'Dose amount description']/value/value as dose_amount,
   b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose timing description']/value/value as dose_timing   from EHR e[ehr_id/value='cd8abecd-9925-4313-86af-93aab4930eae']
    contains COMPOSITION a[openEHR-EHR-COMPOSITION.medication_list.v0]
        contains INSTRUCTION b_a[openEHR-EHR-INSTRUCTION.medication_order.v1]
   where b_a/activities[at0001]/description[at0002]/items[at0173, 'Dose amount description']/value/value='Dose Amount'
```

NB. Although node predicates are supported in the WHERE clause, proper indexing might be required to keep good performance on large dataset.

The test server is now updated with this fix for test. Please check. Thanks

**Comment by chevalleyc** [comment]
included in v1.2.0
